src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -131,6 +139,32 @@ public void close() {
     }
   }
 
+  /*
+  Get/Post DRS Object
+   */
+  /**
+   * Look up the DRS object for a DRS object ID.
+   *
+   * @param drsObjectId the object ID to look up
+   * @param drsPassportRequestModel includes RAS passport, used for authorization and 'expand' var -
+   *     if expand is false and drsObjectId refers to a bundle, then the returned array contains
+   *     only those objects directly contained in the bundle
+   * @return the DRS object for this ID
+   * @throws IllegalArgumentException if there is an issue with the object id
+   * @throws SnapshotNotFoundException if the snapshot for the DRS object cannot be found
+   * @throws TooManyRequestsException if there are too many concurrent DRS lookup requests
+   */
+  public DRSObject lookupObjectByDrsIdPassport(
+      String drsObjectId, DRSPassportRequestModel drsPassportRequestModel) {
+    try (DrsRequestResource r = new DrsRequestResource()) {
+      Snapshot snapshot = lookupSnapshotForDRSObject(drsObjectId);
```

Comment:
Given the optimization that Trevyn just had to do: we should use the cached object I think